### PR TITLE
Add support for the "String" ABI type

### PR DESCRIPTION
**Pull request by dforsten**
In the course of implementing the ERC777 Token Standard using pwasm (available on GitHub at https://github.com/lab10-coop/pwasm-erc777) and produce the same JSON ABI output as the Solidity reference implementation (available on GitHub at https://github.com/0xjac/ERC777) I needed to implement the AbiType trait for "String" in the pwasm-abi repository.

The "String" type also has been added to the "types" module for consistency with other types supported by the pwasm-abi crate.

### CODE REVIEW

**Comment by dforsten** [comment]
[clabot:check]

**Comment by dforsten** [comment]
Unit Test committed and checks passed!

File: src/eth/tests.rs [hunk 1]
```diff
@@ -227,3 +227,33 @@ fn padding_test_i64() {
 	let mut stream = ::eth::Stream::new(&sample);
 	assert_eq!(stream.pop::<i64>().unwrap_err(), Error::InvalidPadding);
 }
+
+
+#[test]
+fn string_encode_decode() {
+	let test_string = String::from("Parity Röcks!");
```
Comment (fckt) [inline src/eth/tests.rs]: 😅

**Comment by fckt** [comment]
To make this PR ideal, it would be good add an acceptance test (for `pwasm_abi_derive`), as we do it for arrays, for example:

https://github.com/paritytech/pwasm-abi/blob/master/tests/src/arrays.rs

**Comment by dforsten** [comment]
Acceptance test added!

**Comment by NikVolf** [comment]
Thanks!

**Comment by dforsten** [comment]
Thanks, Nikolay! :-)

**Pull request merged**